pkgs/by-name/wl/wl-clicker/package.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  wayland-scanner,
+  wlr-protocols,
+  wayland,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wl-clicker";
+  version = "0.3.1";
+
+  nativeBuildInputs = [ wayland-scanner ];
+  buildInputs = [
+    wlr-protocols
+    wayland
+  ];
+
+  src = fetchFromGitHub {
+    owner = "phonetic112";
+    repo = "wl-clicker";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-0ddi383ilxqm0nvn05asn6m77sgi96fw07np95wx089a97b0mr8y";
+  };
+
+  postPatch = ''
+    sed -i 's|/usr|${wlr-protocols}|g' Makefile
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    install build/wl-clicker $out/bin/wl-clicker
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Wayland autoclicker";
+    longDescription = "Script for auto clicking at incredibly high speeds - user must be a part of `input` group to run.";
+    homepage = "https://github.com/phonetic112/wl-clicker";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.Flameopathic ];
+    mainProgram = "wl-clicker";
+    platforms = lib.platforms.linux;
+    broken = (stdenv.isLinux && stdenv.isAarch64) || stdenv.isDarwin;
```

Comment:
If excluding darwin from platforms was intended, then you can omit darwin here. Either specify platforms.unix in platforms or simplify the `broken =` line.

Reply:
it was; understood. thank you for taking me through this process in such detail, sorry for the nubile errors. adjusted.